code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceConsumeMapDaoImpl.java

```diff
@@ -66,4 +66,15 @@ public ServiceConsumeMap findNonRemovedMap(long serviceId, long consumedServiceI
                                 .and(SERVICE_CONSUME_MAP.REMOVED.isNull())).fetchInto(ServiceConsumeMapRecord.class);
     }
 
+    @Override
+    public List<? extends ServiceConsumeMap> findConsumingMapsToRemove(long serviceId) {
+        return create()
+                .selectFrom(SERVICE_CONSUME_MAP)
+                .where(
+                        SERVICE_CONSUME_MAP.CONSUMED_SERVICE_ID.eq(serviceId)
+                                .and((SERVICE_CONSUME_MAP.REMOVED.isNull().
+                                        or(SERVICE_CONSUME_MAP.STATE.eq(CommonStatesConstants.REMOVING))))).
```

Comment:
Want to make sure I understand this query, you want to get maps "where removed is null or state is removing."
Is that right?
Seems to me like it should be "where removed is null and state is not removing"?